src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Serialization/ObjectReader.cs

```diff
@@ -232,15 +232,22 @@ public object ReadValue()
                 return ReadStringValue(code);
             case TypeCode.DateTime:
                 return DateTime.FromBinary(ReadInt64());
-            case TypeCode.Array:
-            case TypeCode.Array_0:
-            case TypeCode.Array_1:
-            case TypeCode.Array_2:
-            case TypeCode.Array_3:
-                return ReadArray(code);
+
+            default:
+                throw ExceptionUtilities.UnexpectedValue(code);
+        }
+    }
+
+    public Encoding? ReadEncoding()
+    {
+        var code = (TypeCode)ReadByte();
+        switch (code)
+        {
+            case TypeCode.Null:
+                return null;
 
             case TypeCode.EncodingName:
-                return Encoding.GetEncoding(ReadString());
+                return Encoding.GetEncoding(ReadString()!);
```

Comment:
>ReadString [](http://example.com/codeflow?start=44&length=10)

Why not use ReadRequiredString here too?

Reply:
didn't exist at teh time.  fixed up though now :)